components/organization/settings/settingsSection.js

```diff
@@ -1,31 +1,36 @@
 import { useEffect, useState } from 'react'
-import { CircularProgress } from '@chakra-ui/core'
+import { CircularProgress, Text } from '@chakra-ui/core'
+import { useRouter } from 'next/router'
 
 import Section from '../../common/section'
+import Card from '../../common/card'
 import UnderlinedHeading from '../../common/underlinedHeading'
 
 import BillingInformationSection from './billingInformationSection'
 import DescriptionSection from './descriptionSection'
 import PublicallyGiveSection from './publicallyGiveSection'
-import { getOrganization } from '../../../client'
 
-import { useLocalStorage } from '../../../utils/useLocalStorage'
-import { localStorageOrgKey } from '../../../utils/constants'
+import { useAuth } from '../../../utils/useAuth'
 
 const OrgSettingsSection = () => {
   const [orgLoading, setOrgLoading] = useState(true)
-  const [org, setOrg] = useState(undefined)
-
-  const [currentOrgId, _] = useLocalStorage(localStorageOrgKey, '') // eslint-disable-line  
+  const [allowedToView, setAllowedToView] = useState(true)
+  const { org: currentOrg, getOrg } = useAuth()
+  const router = useRouter()
 
   async function fetchOrg () {
+    if (!router.query || !router.query.organizationId) return
+    const orgId = router.query.organizationId
     try {
-      const res = await getOrganization({ orgId: currentOrgId })
-      setOrg({
-        id: currentOrgId,
-        ...res.organization
-      })
-    } catch (e) {} finally {
+      const { isOrgAdmin } = await getOrg({ orgId })
```

Comment:
check whether the user is admin again on settings page just in case someone force navs there